src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1081,62 +1081,52 @@ public StringBuilder Append(char[] value)
             return this;
         }
 
-        // Append joined values with a separator between each value.
-        public unsafe StringBuilder AppendJoin<T>(char separator, params T[] values)
-        {
-            // Defer argument validation to the internal function
-            return AppendJoinCore(&separator, 1, values);
-        }
+        
+        #region AppendJoin
 
-        public unsafe StringBuilder AppendJoin<T>(string separator, params T[] values)
+        public unsafe StringBuilder AppendJoin(string separator, params object[] values)
         {
             separator = separator ?? string.Empty;
             fixed (char* pSeparator = separator)
             {
-                // Defer argument validation to the internal function
                 return AppendJoinCore(pSeparator, separator.Length, values);
             }
         }
 
-        public unsafe StringBuilder AppendJoin<T>(char separator, IEnumerable<T> values)
+        public unsafe StringBuilder AppendJoin<T>(string separator, IEnumerable<T> values)
         {
-            // Defer argument validation to the internal function
-            return AppendJoinCore(&separator, 1, values);
+            separator = separator ?? string.Empty;
+            fixed (char* pSeparator = separator)
+            {
+                return AppendJoinCore(pSeparator, separator.Length, values);
+            }
         }
 
-        public unsafe StringBuilder AppendJoin<T>(string separator, IEnumerable<T> values)
+        public unsafe StringBuilder AppendJoin(string separator, params string[] values)
```

Comment:
If we are to match `String.Join` it is (by accident most likely) `value` not `values`. unlike for `object[]`. @terrajobst do we care? can we be values in both cases?